crates/matrix-sdk/src/widget/machine/incoming.rs

```diff
@@ -49,23 +49,34 @@ pub(crate) enum IncomingMessage {
     /// This means that the machine previously subscribed to some events
     /// ([`crate::widget::Action::Subscribe`] request).
     MatrixEventReceived(Raw<AnyTimelineEvent>),
+
+    /// The `MatrixDriver` notified the `WidgetMachine` of a change in room
+    /// state.
+    ///
+    /// This means that the machine previously subscribed to some events
+    /// ([`crate::widget::Action::Subscribe`] request).
+    StateUpdateReceived(Vec<Raw<AnyStateEvent>>),
 }
 
 pub(crate) enum MatrixDriverResponse {
     /// Client acquired capabilities from the user.
-    ///
-    /// A response to an `Action::AcquireCapabilities` command.
+    /// A response to a `MatrixDriverRequestData::AcquireCapabilities` command.
```

Comment:
nit: here and below, use the cargo docs link notation, so we get links for free when rendered:

```rust
/// A response to a [`MatrixDriverRequestData::AcquireCapabilities`] command.
```